addons/account_analytic_analysis/account_analytic_analysis.py

```diff
@@ -93,486 +68,497 @@ def product_id_change(self, cr, uid, ids, product, uom_id, qty=0, name='', partn
-                            JOIN product_product pp \
-                                ON (account_analytic_line.product_id = pp.id) \
-                            JOIN product_template pt \
-                                ON (pp.product_tmpl_id = pt.id) \
-                            JOIN account_analytic_account a \
-                                ON (a.id=account_analytic_line.account_id) \
-                            JOIN hr_timesheet_invoice_factor hr \
-                                ON (hr.id=a.to_invoice) \
-                        WHERE account_analytic_line.account_id IN %s \
-                            AND a.to_invoice IS NOT NULL \
-                            AND account_analytic_journal.type IN ('purchase', 'general')
-                        GROUP BY account_analytic_line.account_id""",(parent_ids,))
-                    for account_id, sum in cr.fetchall():
-                        res[account_id][f] = round(sum, dp)
-        return res
+                                AND a.to_invoice IS NOT NULL \
+                                AND account_analytic_journal.type IN ('purchase', 'general')
+                            GROUP BY account_analytic_line.account_id""",(parent_ids,))
+                        for account_id, sum in self._cr.fetchall():
+                            self.ca_theorical = round(sum, dp)
+        
+    @api.multi
+    @api.depends('ca_invoiced', 'fix_price_invoices','timesheet_ca_invoiced')
+    def _ca_invoiced_calc(self):
+        for account in self:
+            account.ca_invoiced =  0.0
 
-    def _ca_invoiced_calc(self, cr, uid, ids, name, arg, context=None):
-        res = {}
-        res_final = {}
-        child_ids = tuple(ids) #We don't want consolidation for each of these fields because those complex computation is resource-greedy.
-        for i in child_ids:
-            res[i] =  0.0
-        if not child_ids:
-            return res
-
-        if child_ids:
+        if self.ids:
             #Search all invoice lines not in cancelled state that refer to this analytic account
-            inv_line_obj = self.pool.get("account.invoice.line")
-            inv_lines = inv_line_obj.search(cr, uid, ['&', ('account_analytic_id', 'in', child_ids), ('invoice_id.state', 'not in', ['draft', 'cancel']), ('invoice_id.type', 'in', ['out_invoice', 'out_refund'])], context=context)
-            for line in inv_line_obj.browse(cr, uid, inv_lines, context=context):
+            inv_line_obj = self.env["account.invoice.line"]
+            inv_lines = inv_line_obj.search(['&', ('account_analytic_id', 'in', self.ids), ('invoice_id.state', 'not in', ['draft', 'cancel']), ('invoice_id.type', 'in', ['out_invoice', 'out_refund'])])
+            for line in inv_lines:
                 if line.invoice_id.type == 'out_refund':
-                    res[line.account_analytic_id.id] -= line.price_subtotal
+                    self.ca_invoiced -= line.price_subtotal
```

Comment:
No. Be careful to use Active Record Pattern while writing the value. As Each assignment will trigger a write action on the database !!!